front/components/assistant/GalleryAssistantPreviewContainer.tsx

```diff
@@ -27,172 +25,52 @@ interface GalleryAssistantPreviewContainerProps {
   ) => void;
 }
 
-const useAssistantUpdate = (
-  agentConfiguration: LightAgentConfigurationType,
-  owner: WorkspaceType,
-  sendNotification: (notification: NotificationType) => void,
-  onSuccess: (isAdded: boolean) => void,
-  onError: () => void
-) => {
-  const updateAssistant = async (
-    action: "added" | "removed",
-    flow: AssistantPreviewFlow,
-    listStatus: AgentUserListStatus,
-    scope: "workspace" | "published"
-  ) => {
-    const isAdding = action === "added";
-    const url =
-      flow === "workspace"
-        ? `/api/w/${owner.sId}/assistant/agent_configurations/${agentConfiguration.sId}/scope`
-        : `/api/w/${owner.sId}/members/me/agent_list_status`;
-
-    const fullAssistantRes = await fetch(
-      `/api/w/${owner.sId}/assistant/agent_configurations/${agentConfiguration.sId}`,
-      {
-        method: "GET",
-        headers: {
-          "Content-Type": "application/json",
-        },
-      }
-    );
-
-    if (!fullAssistantRes.ok) {
-      throw new Error(
-        (await fullAssistantRes.json()).error.message ?? "Error fetching"
-      );
-    }
-
-    const body:
-      | { scope: "workspace" | "published" }
-      | PostAgentListStatusRequestBody =
-      flow === "workspace"
-        ? {
-            scope,
-          }
-        : {
-            agentId: agentConfiguration.sId,
-            listStatus,
-          };
-
-    try {
-      const response = await fetch(url, {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-        },
-        body: JSON.stringify(body),
-      });
-
-      if (!response.ok) {
-        throw new Error((await response.json()).error.message);
-      }
-
-      sendNotification({
-        title: `Assistant ${isAdding ? "added to" : "removed from"} ${
-          flow === "workspace" ? "Workspace" : "your"
-        } list`,
-        type: "success",
-      });
-
-      onSuccess(isAdding);
-    } catch (error) {
-      sendNotification({
-        title: `Error ${isAdding ? "adding" : "removing"} Assistant`,
-        description: error instanceof Error ? error.message : String(error),
-        type: "error",
-      });
-      onError();
-    }
-  };
-
-  return updateAssistant;
-};
-
 export function GalleryAssistantPreviewContainer({
```

Comment:
I'll remove this proxy component in a follow up PR when removing the `<AssistantPreview>`.